barretenberg/cpp/src/barretenberg/ultra_honk/databus_composer.test.cpp

```diff
@@ -87,13 +56,41 @@ TEST_F(DataBusComposerTests, SingleCircuit)
 
     auto builder = proof_system::GoblinUltraCircuitBuilder{ op_queue };
 
+    // Create a general test circuit
     generate_test_circuit(builder);
 
+    // Add some values to calldata and store the corresponding witness index
+    std::array<FF, 5> calldata_values = { 7, 10, 3, 12, 1 };
+    std::vector<uint32_t> calldata_value_indices;
+    for (auto& val : calldata_values) {
+        calldata_value_indices.emplace_back(builder.add_public_calldata(val));
+    }
+
+    // Add some read index values and store the corresponding witness index
+    std::array<uint32_t, 2> read_index_values = { 1, 4 };
+    std::vector<uint32_t> read_indices;
+    for (auto& val : read_index_values) {
+        read_indices.emplace_back(builder.add_variable(val));
+    }
+
+    // Create some calldata lookup gates
+    for (size_t i = 0; i < read_indices.size(); ++i) {
+        uint32_t read_idx = read_indices[i];
+        uint32_t value_idx = calldata_value_indices[read_index_values[i]];
```

Comment:
I think the way it is right now, the verification key would change depending on read_index_values. If someone were to take this test and use databus based on this pattern it could be dangerous. Ideally we'd create a new variable with the same value here.

Reply:
updated so that the witness used in the lookup is unique as we discussed offline